### PR TITLE
Add option to have truncation_index instead of truncation_threshold

### PR DESCRIPTION
**Pull request by gmegh**

### CODE REVIEW

File: python/lsst/ts/ofc/ofc_data/ofc_data.py [hunk 1]
```diff
@@ -729,9 +729,13 @@ def configure_controller(self) -> None:
...
-        if "truncation_threshold" not in self.controller:
+        if (
+            "truncation_threshold" not in self.controller
+            and "truncation_index" not in self.controller
+        ):
             raise ValueError(
-                "Required key 'truncation_threshold' is missing in the controller configuration."
+                "Required keys 'truncation_threshold' or 'truncation_index' are missing "
+                "in the controller configuration."
             )
```
[the remark this answers is not in the document]
Comment (jbkalmbach) [reply python/lsst/ts/ofc/ofc_data/ofc_data.py:739]: Should also add tests in the tests file that verify the desired behavior when both or None are set.

**Review by jbkalmbach — APPROVED** [review]
Looks okay but I left some suggestions for logic checking when both index and threshold are set to make things clearer about which values are used when.

File: python/lsst/ts/ofc/state_estimator.py [hunk 2]
```diff
@@ -135,6 +136,14 @@ def dof_state(
 
         # Compute the pseudo-inverse of the sensitivity matrix
         # rcond sets the truncation of different modes.
+        # If rcond is None, it is computed from the singular values
+        # of the sensitivity matrix, using the truncation index as reference.
+        if self.rcond is None:
+            if self.truncate_index is None:
+                raise ValueError("Truncation index not set in the controller.")
```
Comment (jbkalmbach) [inline python/lsst/ts/ofc/state_estimator.py:143]: I think this should mention that they are both `None` instead of assuming which one the user wants to set.

**Review by jbkalmbach — APPROVED** [review]
Looks good. Minor comment.
